exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -176,6 +181,23 @@ public void setAutoRead(boolean enableAutoRead) {
   }
 
   /**
+   * Sets the client name.
+   *
+   * If not set, default is {@code DrillClient#DEFAULT_CLIENT_NAME}.
+   *
+   * @param name the client name
+   *
+   * @throws IllegalStateException if called after a connection has been established.
+   * @throws NullPointerException if client name is null
```

Comment:
DrillClient have a default name, so name won't be null. Setting a null name though will cause an exception...